tests/windows_test.go

```diff
@@ -0,0 +1,199 @@
+						Hyperv: &v1.HypervTimer{},
+					},
+				},
+				Firmware: &v1.Firmware{UUID: firmware},
+				Resources: v1.ResourceRequirements{
+					Requests: k8sv1.ResourceList{
+						k8sv1.ResourceMemory: resource.MustParse("4096Mi"),
+					},
+				},
+				Devices: v1.Devices{
+					Disks: []v1.Disk{
+						{
+							Name:       windowsDisk,
+							VolumeName: windowsDisk,
+							DiskDevice: v1.DiskDevice{Disk: &v1.DiskTarget{Bus: "sata"}},
+						},
+					},
+				},
+			},
+			Volumes: []v1.Volume{
+				{
+					Name: windowsDisk,
+					VolumeSource: v1.VolumeSource{
+						Ephemeral: &v1.EphemeralVolumeSource{
+							PersistentVolumeClaim: &k8sv1.PersistentVolumeClaimVolumeSource{
+								ClaimName: tests.DiskWindows,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	tests.BeforeAll(func() {
+		windowsPv, err := virtClient.CoreV1().PersistentVolumes().Get(tests.DiskWindows, metav1.GetOptions{})
+		if err != nil {
+			Skip(fmt.Sprintf("Skip Windows tests that requires PVC %s", tests.DiskWindows))
+		}
+		windowsPv.Spec.ClaimRef = nil
+		_, err = virtClient.CoreV1().PersistentVolumes().Update(windowsPv)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	BeforeEach(func() {
+		tests.BeforeTestCleanup()
+	})
+
+	It("should success to start", func() {
+		vm, err := virtClient.VM(tests.NamespaceTestDefault).Create(windowsVm)
+		Expect(err).To(BeNil())
+		tests.WaitForSuccessfulVMStartWithTimeout(vm, 180)
+	}, 300)
+
+	It("should success to stop", func() {
+		By("Starting a VM")
+		vm, err := virtClient.VM(tests.NamespaceTestDefault).Create(windowsVm)
+		Expect(err).To(BeNil())
+		tests.WaitForSuccessfulVMStartWithTimeout(vm, 180)
+
+		By("Stopping the VM")
+		err = virtClient.VM(tests.NamespaceTestDefault).Delete(vm.Name, &metav1.DeleteOptions{})
+		Expect(err).To(BeNil())
+	}, 300)
+
+	Context("with winrm connection", func() {
+		var winrmcliPod *k8sv1.Pod
+		var cli string
+
+		BeforeEach(func() {
+			tests.BeforeTestCleanup()
+			winrmcliPod = &k8sv1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: winrmCli},
+				Spec: k8sv1.PodSpec{
+					Containers: []k8sv1.Container{
+						{
+							Name:    winrmCli,
+							Image:   fmt.Sprintf("%s/%s:%s", tests.KubeVirtRepoPrefix, winrmCli, tests.KubeVirtVersionTag),
+							Command: []string{"sleep"},
+							Args:    []string{"3600"},
+						},
+					},
+				},
+			}
+			winrmcliPod, err = virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Create(winrmcliPod)
+			Expect(err).ToNot(HaveOccurred())
+
+			vm, err := virtClient.VM(tests.NamespaceTestDefault).Create(windowsVm)
+			Expect(err).To(BeNil())
+			tests.WaitForSuccessfulVMStartWithTimeout(vm, 180)
+
+			vm, err = virtClient.VM(tests.NamespaceTestDefault).Get(vm.Name, metav1.GetOptions{})
+			cli = fmt.Sprintf("%s -hostname %s -username \"%s\" -password \"%s\"",
+				winrmCliCmd,
+				vm.Status.Interfaces[0].IP,
+				windowsVmUser,
+				windowsVmPassword,
+			)
+		})
+
+		XIt("should have correct UUID", func() {
```

Comment:
The X means "skip", right?

Reply:
Yes, I still trying to make work exec command via API, so I disable this tests for skip them locally.